src/ansys/simai/core/data/models.py

```diff
@@ -38,6 +114,43 @@ class ModelConfiguration:
     simulation_volume: Dict[str, Any] = None
     project_id: str = None
 
+    @property
+    def domain_of_analysis(self) -> DomainOfAnalysis | None:
+        """The Domain of Analysis of the model configuration."""
+        if not self.simulation_volume:
+            return None
+
+        doa = DomainOfAnalysis()
+
+        doa.Length = self._get_doa_axis("X")
+        doa.Width = self._get_doa_axis("Y")
+        doa.Height = self._get_doa_axis("Z")
+
+        return doa
+
+    @domain_of_analysis.setter
+    def domain_of_analysis(self, doa: DomainOfAnalysis):
+        """Sets the Domain of Analysis in the configuration."""
+
+        if self.simulation_volume is None:
+            self.simulation_volume = {}
+
+        self.simulation_volume["X"] = self._set_doa_axis(doa.Length, "Length")
+        self.simulation_volume["Y"] = self._set_doa_axis(doa.Width, "Width")
+        self.simulation_volume["Z"] = self._set_doa_axis(doa.Height, "Height")
+
+    def _get_doa_axis(self, rel_pos: str) -> DomainAxisDefinition:
+        pos = self.simulation_volume.get(rel_pos)
+        return DomainAxisDefinition(position=pos["type"], value=pos["value"], length=pos["length"])
+
+    def _set_doa_axis(self, fld: DomainAxisDefinition, param: str) -> Dict[str, Any]:
+        """Serializes a DomainAxisDefinition to the required format for the server."""
+        if fld is None:
+            raise InvalidArguments(
+                f"Empty parameter {param} found when setting the Domain of Analysis. All the parameters should be filled in."
```

Comment:
f"Empty parameter {param} found when setting the Domain of Analysis. All parameters should be set."